### PR TITLE
Link account command & tracking

### PR DESCRIPTION
**Pull request by shreyanshss7**

### CODE REVIEW

File: fyle_teams_app/views/handlers/command_handlers.py [hunk 1]
```diff
@@ -60,3 +70,28 @@ async def handle_fyle_unlink_account(self, turn_context: TurnContext) -> JsonRes
         await turn_context.send_activity(message)
 
         return JsonResponse({})
+
+
+    async def handle_link_fyle_account(self, turn_context: TurnContext, user_id: str, team_id: str) -> JsonResponse:
+
```
Comment (gokulk16) [inline fyle_teams_app/views/handlers/command_handlers.py:76]: need not have these many line breaks after every line. Please use line breaks only when there is a logical separation

**Review by gokulk16 — CHANGES_REQUESTED** [review]

File: fyle_teams_app/models/users.py [hunk 2]
```diff
@@ -128,9 +128,18 @@ def track_fyle_account_linked(user, fyle_profile: Dict) -> None:
 
         tracking.track_event(user.email, 'Fyle Account Linked To Teams', event_data)
 
+    @staticmethod
+    def track_fyle_account_unlinked(event_data) -> None:
```
Comment (shreyanshss7) [inline fyle_teams_app/models/users.py:132]: TODO: All the trackings should be in a separate module/file. Will take care of this in a separate PR.

File: fyle_teams_app/models/users.py [hunk 3]
```diff
@@ -128,9 +128,18 @@ def track_fyle_account_linked(user, fyle_profile: Dict) -> None:
...
+
+        email = event_data['email']
+
```
Comment (gokulk16) [inline fyle_teams_app/models/users.py:135]: why is this unnecesary line breaks?

File: fyle_teams_app/views/authorisation/team.py [hunk 4]
```diff
@@ -57,4 +57,8 @@ async def bot_uninstalled(turn_context: TurnContext):
         # Remove user
         await User.remove_user(user)
 
+        user_details = await TeamsInfo.get_member(turn_context, user_id)
+
```
Comment (gokulk16) [inline fyle_teams_app/views/authorisation/team.py:61]: line breaks is not for every line :fml:

File: fyle_teams_app/views/handlers/action_handlers.py [hunk 5]
```diff
@@ -87,3 +93,18 @@ async def handle_approve_report(self, turn_context: TurnContext, action_details:
...
+            'team_user_id': user.team_user_id,
+            'fyle_user_id': user.fyle_user_id,
+            'email': user.email,
+            'team_id': user.team_id,
+            'report_id': report['id'],
+            'org_id': report['org_id']
+        }
+
+        tracking = Tracking(user.email)
+
```
Comment (gokulk16) [inline fyle_teams_app/views/handlers/action_handlers.py:109]: remove line break

File: fyle_teams_app/views/handlers/command_handlers.py [hunk 6]
```diff
@@ -30,25 +34,41 @@ async def handle_commands(self, turn_context: TurnContext) -> JsonResponse:
...
-        user_id = turn_context_dict['from_property']['id']
+
+    async def handle_unlink_fyle_account(self, turn_context: TurnContext, user_id: str, team_id: str) -> JsonResponse:
 
         user = await User.get_by_id(user_id)
 
-        if user.fyle_user_id is None:
+        fyle_user_id = user.fyle_user_id
+
+        if fyle_user_id is None:
```
Comment (gokulk16) [inline fyle_teams_app/views/handlers/command_handlers.py:56]: remove linebreak

File: fyle_teams_app/views/handlers/command_handlers.py [hunk 7]
```diff
@@ -30,25 +34,41 @@ async def handle_commands(self, turn_context: TurnContext) -> JsonResponse:
...
+
+            user_details_to_track = {
+                'user_id': user_id,
+                'team_id': team_id,
+                'fyle_user_id': fyle_user_id,
+                'email': user.email
+            }
+
+            processing_message = 'Hey we are unlinking your Fyle account with Microsft Teams, we will send a message once it is done'
+
```
Comment (gokulk16) [inline fyle_teams_app/views/handlers/command_handlers.py:69]: remove linebreak

File: fyle_teams_app/views/handlers/command_handlers.py [hunk 8]
```diff
@@ -30,25 +34,41 @@ async def handle_commands(self, turn_context: TurnContext) -> JsonResponse:
...
         else:
+
+            user_details_to_track = {
+                'user_id': user_id,
+                'team_id': team_id,
+                'fyle_user_id': fyle_user_id,
+                'email': user.email
+            }
+
+            processing_message = 'Hey we are unlinking your Fyle account with Microsft Teams, we will send a message once it is done'
```
Comment (gokulk16) [inline fyle_teams_app/views/handlers/command_handlers.py:68]: can remove 'Hey'

**Review by gokulk16 — CHANGES_REQUESTED** [review]

File: fyle_teams_app/views/handlers/action_handlers.py [hunk 9]
```diff
@@ -71,6 +73,10 @@ async def handle_approve_report(self, turn_context: TurnContext, action_details:
                     report = platform.v1beta.approver.reports.approve(report_id)
                     report = report['data']
                     report_message = '**Expense Report Approved** 🚀 '
+
+                    # Track report approved
```
Comment (gokulk16) [inline fyle_teams_app/views/handlers/action_handlers.py:77]: avoid these kind of code comments where it is not saying anything beyond the function name.
can remove.

File: fyle_teams_app/views/handlers/action_handlers.py [hunk 10]
```diff
@@ -10,6 +10,8 @@
 from fyle_teams_app.models import User
 from fyle_teams_app.libs import logger, fyle_utils, assertions
 from fyle_teams_app.ui.cards import notifications as notification_cards
+from fyle_teams_app.libs.tracking import Tracking
+
```
Comment (gokulk16) [inline fyle_teams_app/views/handlers/action_handlers.py:15]: 3 line breaks? Why Shreyansh Why

File: fyle_teams_app/models/users.py [hunk 11]
```diff
@@ -139,5 +145,4 @@ def track_bot_installed(user_details: TeamsChannelAccount):
         }
 
         tracking = Tracking(user_details.email)
-
```
Comment (gokulk16) [inline fyle_teams_app/models/users.py:142]: remove line break

File: fyle_teams_app/views/authorisation/team.py (hunk 4 above)
Comment (gokulk16) [reply fyle_teams_app/views/authorisation/team.py:61]: please remove.

Comment (gokulk16) [reply fyle_teams_app/views/authorisation/team.py:61]: having line break to call a function is fine. can also group a function call's related parameters also together.
example this line.
`user_details` is only used in the next line(a function call). It's fine to group these 2 together.

File: fyle_teams_app/models/users.py (hunk 11 above)
Comment (gokulk16) [reply fyle_teams_app/models/users.py:142]: the above `tracking`  object is used only for the next line. So can have it together without line breaks.

**Review by gokulk16 — APPROVED** [review]

File: fyle_teams_app/views/authorisation/team.py (hunk 4 above)
Comment (shreyanshss7) [reply fyle_teams_app/views/authorisation/team.py:60]: Agreed

File: fyle_teams_app/views/handlers/action_handlers.py (hunk 10 above)
Comment (shreyanshss7) [reply fyle_teams_app/views/handlers/action_handlers.py:15]: my bad here

**Pull request merged**